es5-persistence/src/main/java/com/netflix/conductor/dao/es5/index/ElasticSearchRestDAOV5.java

```diff
@@ -786,7 +790,143 @@ public String get(String workflowInstanceId, String fieldToGet) {
         return workflowIds.getResults();
     }
 
-    private void indexObject(final String index, final String docType, final String docId, final Object doc) {
+    /**
+     *   Prune tasks based on Ids
+     * @param List of tasks that need to be pruned(Usually these are the tasks whose parent workflows were alredy pruned).
+     */
+    @Override
+    public void pruneTasks(List<String> taskIds) {
+        String docType = TASK_DOC_TYPE;
+        // Prune tasks that that belonged to deleted workflows
+        if (taskIds.size() > 0) {
+            BulkRequest bulkRequest = new BulkRequest();
+            for (String taskId:taskIds) {
+                bulkRequest.add(new DeleteRequest(indexName, docType, taskId));
+            }
+            pruneBulkRecords(bulkRequest, docType, taskIds.size(), 0);
+        }
+
+        //Prune all tasks older than 14 days (or) all completed tasks no matter what
+ /*
+        int daysToKeep = config.getPruningDaysToKeep();
+        int batchSize = config.getPruningBatchSize();
+        DateTime dateTime = new DateTime();
+        QueryBuilder taskQuery = QueryBuilders.boolQuery()
+                                    .must(QueryBuilders.boolQuery()
+                                            .should(QueryBuilders.termQuery("status", "COMPLETED"))
+                                            .must(QueryBuilders.rangeQuery("updateTime").gt(dateTime.minusDays(daysToKeep)))
+                                    .must(QueryBuilders.boolQuery()
+                                            .must(QueryBuilders.termQuery("status", "FAILED"))
+                                            .must(QueryBuilders.rangeQuery("updateTime").lt(dateTime.minusDays(daysToKeep)))
+                                    );
+
+        pruneDocs(indexName, taskQuery, TASK_DOC_TYPE, batchSize, Collections.singletonList("endTime:ASC"));
+  */
+    }
+
+    /**
+     *   Prune workflows that are archived
+     * @return list of workflow Ids that were pruned.
+     */
+    @Override
+    public List<String> pruneWorkflows() {
+        // Prune oldest archived workflows older than 7 days by batch size
+        int daysToKeep = config.getPruningDaysToKeep();
+        DateTime dateTime = new DateTime();
+        //Prune all workflows older than 14 days (or) all archived & completed workflows no matter what
+        int daysForDebug = 14;
+        int daysWaitingAllowed = 180;
```

Comment:
Since the current limit is 180 days, please increase this to 181 days.

Reply:
Done